### ISSUE TITLE
complemented exception var of `try {} catch {}` is collided with variable declaration identifier with const or let

### ISSUE THREAD

**Issue by Shinyaigeek** [swc-project/swc#4632]
### Describe the bug

`try {} catch {}`'s exception var cannot be omitted in es2018 and lower es versions, so `swc` complements this exception var with `private_ident!` macro. (FYI: https://github.com/swc-project/swc/blob/main/crates/swc_ecma_transforms_compat/src/es2019/optional_catch_binding.rs#L23)

if a developer already uses `e` identifier, such as declaration `function e() {}` or calling `log(e)`, exception var's `e` name will be renamed in the `hygine` process. However, declaration `e` with const or let is not reckon as used by the `UsageAnalyzer` in `hygine`, so if `const e = ~~` is in `catch` block, variable name collision happens and application break.

### Input code

```typescript
try {} catch {
    const e = hoge
}
```

### Config

```json
{
  "jsc": {
    "parser": {
      "syntax": "typescript",
      "tsx": false
    },
    "target": "es2016",
    "loose": false,
    "minify": {
      "compress": false,
      "mangle": false
    }
  },
  "module": {
    "type": "es6"
  },
  "minify": false,
  "isModule": true
}
```

### Playground link

https://play.swc.rs/?version=1.2.179&code=H4sIAAAAAAAAAyspqlSorlVITixJzlCo5lIAguT8vOIShVQFW4WM%2FPRUrloA61TNXSMAAAA%3D&config=H4sIAAAAAAAAA0WNSw6DMAxE7%2BI1i7YLFtyhh4hSg4Lyk8dIjaLcvQGlYmfPPD9X2mFpqZSNgOWcUKKaLy2kJTOsuKw0kaJHq%2FHg1hcjG2tHGK%2FHc%2B61Twk8gImCi24tp8ymkIWBuzJx83%2BydVdIn%2BMM6vXvcs7Ubse4c3gPUOXg9gOKIRNxuAAAAA%3D%3D

### Expected behavior

```javascript
try {} catch (e1) {
    const e = hoge;
}
```

### Actual behavior

```javascript
try {} catch (e) {
    // this will break
    const e = hoge;
}
```

### Version

1.2.182

**Comment by magic-akari** [comment]
It has been fixed in 1.2.204. maybe #4952.

**Comment by kdy1** [comment]
Thanks!
